File: src/main/java/com/lessmarkup/framework/system/RequestContextHolder.java
/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */
package com.lessmarkup.framework.system;

import com.lessmarkup.interfaces.system.RequestContext;

public final class RequestContextHolder {
    
    private static final ThreadLocal contextHolder = new ThreadLocal();
    
    public static void onRequestStarted(RequestContext context) {
        contextHolder.set(context);
    }
    
    public static void onRequestFinished() {
        contextHolder.remove();
    }
    
    public static RequestContext getContext() {
        return (RequestContext) contextHolder.get();
    }
}

File: src/main/scala/com/lessmarkup/framework/system/RequestContextHolder.scala
/*
 * This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
 * If a copy of the MPL was not distributed with this file, You can obtain one at
 * http://mozilla.org/MPL/2.0/.
 */

package com.lessmarkup.framework.system

import com.lessmarkup.interfaces.system.RequestContext

object RequestContextHolder {

  private val contextHolder = new ThreadLocal[RequestContext]

  def onRequestStarted(context: RequestContext): Unit = {
    contextHolder.set(context)
  }

  def onRequestFinished(): Unit = {
    contextHolder.remove()
  }

  def getContext: RequestContext = {
    contextHolder.get()
  }
}
